### PR TITLE
Integrated API with Backend and Ensured Non-Exception based Response Flow

### PR DESCRIPTION
**Pull request by RobertLaverick**

### CODE REVIEW

File: app/connectors/util/CustomHttpReader.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import play.api.http.Status
+import uk.gov.hmrc.http.{HttpErrorFunctions, HttpReads, HttpResponse}
+
+object CustomHttpReader extends HttpReads[HttpResponse] with HttpErrorFunctions with Status {
+  override def read(method: String, url: String, response: HttpResponse): HttpResponse = {
+    response.status match {
+      case status if is2xx(status) => response
+      case LOCKED => recode(INTERNAL_SERVER_ERROR, response)
+      case BAD_GATEWAY => recode(INTERNAL_SERVER_ERROR, response)
+      case _ => response
```
Comment (jhkldev) [inline app/connectors/util/CustomHttpReader.scala:28]: UNAUTHORIZED, NOT_FOUND and BAD_REQUEST need to be recoded here.

Should the is2xx check be removed?

File: servicemanager/include/python2.7 [hunk 2]
```diff
@@ -0,0 +1 @@
+/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7
```
Comment (jhkldev) [inline servicemanager/include/python2.7:1]: Don't think we need this :stuck_out_tongue:

File: test/connectors/util/CustomHttpReaderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import org.scalatest.{FreeSpec, MustMatchers, OptionValues}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatestplus.mockito.MockitoSugar
+import uk.gov.hmrc.http.HttpResponse
+import play.api.http.Status
+import org.scalacheck.Gen
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+class CustomHttpReaderSpec extends FreeSpec with MustMatchers with OptionValues with ScalaFutures with MockitoSugar with ScalaCheckDrivenPropertyChecks {
```
Comment (jhkldev) [inline test/connectors/util/CustomHttpReaderSpec.scala:27]: More tests should be added if UNAUTHORIZED, NOT_FOUND and BAD_REQUEST are recoded.

File: app/connectors/util/CustomHttpReader.scala (hunk 1 above)
Comment (RobertLaverick) [reply app/connectors/util/CustomHttpReader.scala:28]: I'll get on that (including the removal)

File: servicemanager/include/python2.7 (hunk 2 above)
Comment (RobertLaverick) [reply servicemanager/include/python2.7:1]: You saw nothing

File: test/connectors/util/CustomHttpReaderSpec.scala (hunk 3 above)
Comment (RobertLaverick) [reply test/connectors/util/CustomHttpReaderSpec.scala:27]: 👍

**Pull request merged**